contracts/LinniaRecords.sol

```diff
@@ -185,7 +191,7 @@ contract LinniaRecords is Ownable {
     {
         // validate input
         require(dataHash != 0);
-        require(keccak256(dataUri) != keccak256(""));
+        require(keccak256(abi.encodePacked(dataUri)) != keccak256(abi.encodePacked("")));
```

Comment:
maybe also check dataUri != "0x"?

Reply:
I'm not sure why you can't just `require(dataUri != "")`

Reply:
TypeError: Operator != not compatible with types string memory and literal_string ""
        require(dataUri != "");

 dataUri != "0x" could be a new issue.